geode-core/src/main/java/org/apache/geode/security/SecurityManager.java

```diff
@@ -74,8 +74,11 @@ default void init(Properties securityProps) {}
    * @param principal The principal that's requesting the permission
    * @param permission The permission requested
    * @return true if authorized, false if not
+   *
+   * @throw AuthenticationExpiredException if the principal has expired.
    */
-  default boolean authorize(Object principal, ResourcePermission permission) {
+  default boolean authorize(Object principal, ResourcePermission permission)
+      throws AuthenticationExpiredException {
```

Comment:
Not sure if this breaks binary compatibility for User's existing impls. I think it probably does.

Reply:
No, adding exception wouldn't. this is from Oracle java doc site:
13.4.21. Method and Constructor Throws
Changes to the throws clause of methods or constructors do not break compatibility with pre-existing binaries; these clauses are checked only at compile time.

Reply:
Good to know! Thank you